### PR TITLE
Fix/internal modifier parameter

### PR DESCRIPTION
**Pull request by Gabriel-Moraes350**
This PR simulates and tries to fix the build issue due to the following scenario:
- Have a public class with a private method that is going to use the KTestable annotation
- A parameter of this private method is a internal or protected class
- The generated function has a public visibility modifier that triggers an error due to exposing a internal type

A example was created in the kotlin-gradle-kotlin-dsl-example.

### CODE REVIEW

File: catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KModifierClassification.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.aerisconsulting.catadioptre.kotlin
+
+import com.squareup.kotlinpoet.KModifier
+
+enum class KModifierClassification(val kModifier: KModifier, val classification: Int) {
```
[the remark this answers is not in the document]
Comment (ericjesse) [reply catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KModifierClassification.kt:5]:
```suggestion
internal enum class KlassVisibility(val klassModifier: KModifier) {
```

File: catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt [hunk 2]
```diff
@@ -125,6 +121,15 @@ internal class KotlinTestableProcessor : AbstractProcessor() {
         return true
     }
 
+    private fun getVisibilityModifier(classContainer: ImmutableKmWithFlags): KModifier {
```
Comment (ericjesse) [inline catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:124]:
```suggestion
    private fun getVisibilityModifier(classMetadata: ImmutableKmWithFlags): KlassVisibility {
```

File: catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt [hunk 3]
```diff
@@ -277,4 +286,10 @@ internal class KotlinTestableProcessor : AbstractProcessor() {
         testableClassFile.addFunction(functionBuilder.build())
     }
 
+    private fun getLowestVisibility(parametersModifiers: MutableList<KModifier>, classVisibilityModifier: KModifier): KModifier {
+
+        val kModifierWithClassification = parametersModifiers.plus(classVisibilityModifier).map { KModifierClassification.getByKModifier(it) }
```
Comment (ericjesse) [inline catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:291]: (parametersModifiers + classVisibilityModifier).minBy{ it.ordinal } should be shorter as a unique statement of the method.

File: catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt [hunk 4]
```diff
@@ -125,6 +121,15 @@ internal class KotlinTestableProcessor : AbstractProcessor() {
         return true
     }
 
+    private fun getVisibilityModifier(classContainer: ImmutableKmWithFlags): KModifier {
+        return when {
+            classContainer.isPublic -> KModifier.PUBLIC
```
[the remark this answers is not in the document]
Comment (Gabriel-Moraes350) [reply catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:126]: You're right, I didn't think about that, I'm changing it

[the remark this answers is not in the document]
Comment (Gabriel-Moraes350) [reply catadioptre-annotations/src/main/kotlin/io/aerisconsulting/catadioptre/kotlin/KotlinTestableProcessor.kt:126]: I've debugged and created a test for it, and it seems that the change won't be needed. Or at least I was not able to simulate the scenario

**Comment by Gabriel-Moraes350** [comment]
I've updated the PR with the fixes.
I added a couple of tests for the Java generated code, just to verify that the visibility behaviour does not affect Java.
As we don't have the internal modifier in Java, I changed the test a little bit.

**Pull request merged**